### PR TITLE
LPS-173885 Fix label computation

### PR DESCRIPTION
**Pull request by victorg1991**
## Motivation
[Link to ticket](https://issues.liferay.com/browse/LPS-169936)

The goal of this changes is to add a fragment that can be mapped to a multi selection picklist for app pages

## Change summary
- Add a new info field type for multiple select
- Add a new fragment for the multiple select using a list of checkbox, one for each option in the picklist
- Adapt current usages to include the multiple select field

## Test Scenario
- Create a picklist 
- Create an object and add a multiple picklist field
- Create a content page, add a form builder and map it to the creaed object
- Publish and try to create an object entry

### CODE REVIEW

**Comment by victorg1991** [comment]
ci:test:sf

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by victorg1991** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
Backend changes look good 👍

File: modules/apps/fragment/fragment-api/src/main/java/com/liferay/fragment/input/template/parser/FragmentEntryInputTemplateNodeContextHelper.java [hunk 1]
```diff
@@ -356,6 +357,31 @@ else if (infoField.getInfoFieldType() instanceof
 				inputTemplateNode.addAttribute("selectedOptionLabel", label);
 			}
 		}
+		else if (infoField.getInfoFieldType() instanceof
```
Comment (ruben-pulido) [inline modules/apps/fragment/fragment-api/src/main/java/com/liferay/fragment/input/template/parser/FragmentEntryInputTemplateNodeContextHelper.java:360]: You may want to order the else if alphabetically

Comment (victorg1991) [reply modules/apps/fragment/fragment-api/src/main/java/com/liferay/fragment/input/template/parser/FragmentEntryInputTemplateNodeContextHelper.java:360]: done :D

File: modules/apps/fragment/fragment-collection-contributor/fragment-collection-contributor-inputs/src/main/resources/com/liferay/fragment/collection/contributor/inputs/dependencies/multiple-select-from-list/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	);
+	const fieldSet = fragmentElement.querySelector(
+		'.multi-select-list-fieldset'
+	);
+
+	button.addEventListener('click', () => {
+		missionOptions.forEach((option) => {
+			const node = template.content.cloneNode(true);
+
+			const input = node.querySelector('input');
```
Comment (p2kmgcl) [inline modules/apps/fragment/fragment-collection-contributor/fragment-collection-contributor-inputs/src/main/resources/com/liferay/fragment/collection/contributor/inputs/dependencies/multiple-select-from-list/index.js:40]: We are missing the `aria-labelledby` attribute here :smile:

**Review by p2kmgcl — COMMENTED** [review]
I just found a tiny issue, it should be ready to send.

**Comment by victorg1991** [comment]
updated!

**Comment by victorg1991** [comment]
ci:forward

**Comment by victorg1991** [comment]
ci:forward

**Comment by liulukiwi** [comment]
Hi @victorg1991, there's an issue find in this pull, could you check about it?
1. Create a picklist
2. Create an object and add picklist field with created picklist
3. Create a content page, add a form builder and map it to the created object
4. Modify field's label

Actual result:
The label can't be changed.

Expected result:
Able to change field's label

**Comment by victorg1991** [comment]
Thanks @liulukiwi I'll review it

**Comment by liulukiwi** [comment]
Hi @victorg1991, there's another issue find in this pull, could you check about it?

1. Create a picklist with 2 items
2. Create an object and add picklist field with created picklist
3. Create a content page, add a form builder and map it to the created object
4. Publish the content page and view it
5. Go to select options in select from list

Actual result:
The 2 items don't display.

Expected result:
Able to choose the 2 items.

**Comment by victorg1991** [comment]
Hey @liulukiwi I didn't change anything from the picklist, can you try that issue on master? It should be reproducible there too

**Comment by victorg1991** [comment]
ci:forward

**Comment by victorg1991** [comment]
@liulukiwi  fixed the first one that is specific to this pr :p

**Comment by liulukiwi** [comment]
Hi @victorg1991, I can't reproduce the second issue on master at 38777bdf00d43dcb9d978655717a3c5886179e1, l will wait for the new ci results to see if LocalFile.FormContainer#ViewCreatedObjectEntryAfterSubmitForm fails again.

**Comment by victorg1991** [comment]
Hey @liulukiwi  just tried and it failed 🤔 I'll open a ticket for it

**Comment by liulukiwi** [comment]
Hi @victorg1991, I still can't reproduce this issue on master and I'm asking @alessandromalves to help me verify it on master.

**Comment by alessandromalves** [comment]
Hey @victorg1991, I tried to reproduce the second issue here and everything happened as expected.

Verified on master: b38777bdf00d43dcb9d978655717a3c5886179e1
 Server: Tomcat 9.0.71 + MySQL 8.0
AttachedEvidence_2023-02-09

cc.: @liulukiwi

**Comment by victorg1991** [comment]
sorry @liulukiwi  I missunderstand the bug, I fixed it!

**Comment by victorg1991** [comment]
ci:forward

**Comment by liulukiwi** [comment]
Failures are not related, pushed to BChan directly at https://github.com/brianchandotcom/liferay-portal/pull/130055

**Comment by liulukiwi** [comment]
ci:close

**Pull request closed without merging**